mkosi

```diff
@@ -4243,6 +4243,11 @@ def build_stuff(args: CommandLineArguments) -> None:
     setup_package_cache(args)
     workspace = setup_workspace(args)
 
+    # Make sure tmpfiles' aging doesn't interfere with our workspace
+    # while we are working on it.
+    dir_fd = os.open(workspace.name, os.O_RDONLY|os.O_DIRECTORY|os.O_CLOEXEC)
+    fcntl.flock(dir_fd, fcntl.LOCK_EX)
```

Comment:
Should this include `fcntl.LOCK_NB`? If a lock is already held on the directory for some reason, it’s probably better for us to proceed (and risk that the other lock is released before we’re done, making us vulnerable to the tmpfiles cleanup again) than block until the other lock is released (we have no idea when that would be).

Reply:
well, the idea is that if tmpfiles is currently iterating through our directory, then we should wait until it is finished with that before we start using the directory. hence it should be a blocking open